packages/ui/src/components/Switch/SSwitch.vue

```diff
@@ -0,0 +1,100 @@
+<script setup lang="ts">
+interface Props {
+  /**
+   * v-model for two-way data binding
+   */
+  modelValue?: boolean
+  /**
+   * Id for matching switch with label
+   *
+   */
+  id: string
+  /**
+   * Text label for switch
+   *
+   * @default ''
+   */
+  label?: string
+  /**
+   * Attr specfifies whether switch is disabled
+   *
+   * @default false
+   */
+  disabled?: boolean
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  label: '',
+  disabled: false,
+})
+
+const emit = defineEmits<(event: 'update:modelValue', value: boolean) => void>()
+const model = useVModel(props, 'modelValue', emit)
+</script>
+
+<template>
+  <div class="s-switch">
+    <input :id="id" v-model="model" type="checkbox" :disabled="disabled" class="s-switch-button" />
+    <label :for="id" class="s-switch-label">{{ label }}</label>
+  </div>
```

Comment:
According to BEM, isn't it correct to name sub-elements like `s-switch__button` & `s-switch__label`?